### DISCUSSION TITLE
Frient Motion Sensor Pro (MOSZB-140)

### DISCUSSION THREAD

**Discussion by Lyraeone** [Koenkk/zigbee2mqtt#14178] [Devices]
Hi all,

I have recently moved to Zigbee2mqtt and sofar it's good. However, I am missing functions on my Motion Sensors.
The ones I have are supported MOSZB-140 but not all options are being pulled from the device. In the native HOAS->ConbeeII integration you are served with 4 sensors and think they were called like this:

1 Presence Sensor (Heavy movement, only humans mode)
3 Motion Sensors (Sensitive movement, minor movement is detected)

In Zigbee2MQTT you get only Occupancy. I was only using the Presence because the others already trigger when my cat walks by.

I believe my case in point is in this documentation, however I am fairly new to all this :)
https://www.develcoproducts.com/media/1697/moszb-130-14x-technical-manual-motion-sensor-040721.pdf

"
2.1.1 IAS Zone
The IAS Zone sensor is PIR based, sensing moving objects up to 9 meters from the sensor (6 meters when it is
MOSZB-130). The end point is configured as Home Automation profile „IAS Zone“. **The IAS Zone End point is less
sensitive than the Occupancy sensor endpoint**. Therefore if the device is used as an alarm sensor the IAS Zone
end point shall be used.
2.1.2 Occupancy
The occupancy sensor is PIR based, sensing moving objects up to 9 meters from the sensor (6 meters when it is
MOSZB-130). The off-time is adjustable remotely via ZigBee. The end point is configured as Home Automation
profile „Occupancy Sensor“.
"

I guess I am looking for the Home Automation profile „IAS Zone“

Does anyone have any idea on how to proceed?

Regards,

Lyraeone

**Comment by sjorge** [comment]
Looking at the definition, it is already using the IAS Zone for detecting Occupancy and not the msOccupancySensing cluster.

https://github.com/Koenkk/zigbee-herdsman-converters/blob/0c38c907232f121580a07ff780bbb8de45a18e6c/devices/develco.js#L288-L304

We can probably add support for the more sensitive msOccupancySensing but we can't use the generic convertor for that as both will use the `occupancy` attribute in the payload so we'd need a delveco variant for msOccupancySensing calling it something else like `occupancy_high_sensitive` or something, there also seems to be some advanced configuration possible wrt gating the msOccupancySensing behind either temperature or illumination gates, which probably is not going to be very easy to implement.

I'll see if I can get my hands on a MASZB-140.

**Reply by sjorge** [reply]
Managed to find one, I hope, should be here next week ish

**Reply by sjorge** [reply]
@Koenkk I was reading the tech spec a bit more and I have a rough idea how to add the config for temp/lux for the msOccupancySensing stuff.

The problem is we can't just add `fz.occupancy` in addition to `fz.ias_occupancy_alarm_1`  as both will set **occupancy** in the device state.

I am currently leaning to just adding `fz.occupancy` and modifying `fz.ias_occupancy_alarm_1` to support specifying a different payload attribute name via the devices meta. (e.g. `occupancy_ias`). So we en up with `occupancy` AND `occupancy_ias` in the payload.

The alternative is doing the reverse and modifying `fz.occupancy`, but I am not sure what to use as alternative name that one, `occupancy_occupancy` seems worse than `occupancy_ias`,

I do see value in having both the msOccupancySensing and ias alarm1 based occupancy's available for this device though, especially if I can get the configuration for msOccupancySensing to work.

Got a shipping update and should have the device on friday 🤞 and I have time on sunday to work on this.

**Reply by Koenkk** [reply]
> The problem is we can't just add fz.occupancy in addition to fz.ias_occupancy_alarm_1 as both will set occupancy in the device state.

Why is this a problem? I expect that both will update the state consistently.

**Reply by sjorge** [reply]
Learned a bunch more, the msOccupancySensing on the other 3 endpoints are highly configurable. So I'll probable end up enable multiendpoint support. Manually using read/write been playing with the manuSpecific attributes but i'm still doing something wrong for now as it is not behaving as in the technical spec pdf.

**Comment by sjorge** [comment]
It arrived early, there are a to of msOccupancySensing clusters on different endpoints
<img width="194" alt="Screen Shot 2022-09-27 at 19 27 56" src="https://user-images.githubusercontent.com/379665/192595659-56127b11-259e-4b3d-8291-f9181e3e9451.png">

endpoint 34, 40, and 41

Looks like the main endpoint is 35 which also has the ssIasZone which makes sense why we're using this.

I'll start messing around with this on sunday when I have time, I'm guessing 34 might be the high sensitive one and 40 the one gates behind temperature and 41 behind illumance. But that just a guess at the moment.

**Reply by sjorge** [reply]
OK so all 3 endpoints are the same out of the box, just more sensitive than the ssIasZone one.
Each one can be configure with a min temp, max temp, and target level logic.
Each one also seems to have there own Occupied -> Unoccupied timeout. (Tested)

I'll need to rething my original idea to add these. Simply exposing 1/3 and labeling it as 'high_sensitive' or something is the easiest way but  I'm going to try and support the logic configuration, that would be cool as you can then have 3 with user customizable configuration.

**Reply by Lyraeone** [reply]
Sounds great! Thanks for looking into it. I can help test or share any information about Frient devices, my house is littered with them :)

I guess I was using the high sensitive ones then as the anti-cat trigger measure

**Reply by sjorge** [reply]
So far I've been messing around a bit and added the following:
- battery % (calculated from batteryVoltage) (merged, will be in the october release)
- led_control (PR opened, not sure it will make it into the october release)
- occupancy_timeout  (PR opened, not sure it will make it into the october release)

Do you or anyone reading this have a MOSZB-130 or other variant than the 140 ? If so I would be interested the cluster layout of those. A screenshot from the frontend or the databases.db entry would be fine.

**Reply by Lyraeone** [reply]
Cool, these are the ones I have, I don't have the 130 myself..

SPLZB-131 (Smart Plug Mini)
SIRZB-110 (Siren)
AQSZB-110 (Air Quality)
WISZB-120 (Door Sensor Pro)
MOSZB-140 (Motion Sensor Pro)
FLSZB-110 (Flood Sensor)
SMSZB-120 (Smoke Alarm)
HESZB-120 (Heat Alarm)

**Reply by sjorge** [reply]
I’ll leave the MOSZB-130 alone then, I think most of the stuff that I added to the 140 up to now also apply to that one but can’t verify it.

Looks like everything listed above made it to yesterdays release too.

Struggling a bit with the extra occupancy end points at the moment. But made some progress.

**Reply by Lyraeone** [reply]
I updated to the latest release, hoping that the Smoke and Heat Alarm would not error but they also do. I shall open up a thread about it. As for the occupancy, let me know when there is news :0

**Reply by sjorge** [reply]
Not much news, the docs seem incorrect and frient has not responded yet to my e-mail.
So for now only led_control and occupancy_timeout were added.

**Reply by Lyraeone** [reply]
Ok, I did see an error after the update. Might be part of the work in progress?

2022-10-14 12:33:52Failed to configure '0-2-MS-Backdoor', attempt 4 (Error: Read 0x0015bc001a01a882/35 genBasic(["develcoLedControl"], {"sendWhen":"active","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4117,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUP_MANUF_GENERAL_COMMAND') at Endpoint.checkStatus (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:317:28) at Endpoint.read (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:480:22) at Object.configure (/app/node_modules/zigbee-herdsman-converters/devices/develco.js:523:13) at Configure.configure (/app/lib/extension/configure.ts:115:13) at Configure.onMQTTMessage (/app/lib/extension/configure.ts:55:21))

**Reply by sjorge** [reply]
Interesting, what version is the firmware for you? Mine reports `Firmware build date: 20220305 08:08`, I wonder if there are different firmware versions. Shame Develco did not respond to me, I did mail them in English though not in Danish as I do not speak that language.

It does successfully read both led_control (and configuring also works) but not any of supposedly supported attributes on the msOccupancySensing cluster.

Some people did report having issues with the configure getting stuck on reading led_control, but they hit timeouts. You seem to be hitting a UNSUP_MANUF_GENERAL_COMMAND, which is weird as I would expect a 'UNSUPPORTED_ATTRIBUTE' error instead . (cc: @Koenkk having UNSUP_MANUF_GENERAL_COMMAND seems very wrong here right?)

For me it never fails at all, no timeout errors nor UNSUPPORT_ATTRIBUTE or UNSUP_MANUF_GENERAL_COMMAND. All my pair+configure attempts look like the one I captured here: https://github.com/Koenkk/zigbee-herdsman-converters/pull/4731#issuecomment-1267467940 I wonder if there are some subtle firmware differences.
<img width="808" alt="Screen Shot 2022-10-14 at 16 57 56" src="https://user-images.githubusercontent.com/379665/195878415-c6ec5886-c132-4007-a6ca-d68a9a92dc59.png">

<img width="969" alt="Screen Shot 2022-10-14 at 16 58 01" src="https://user-images.githubusercontent.com/379665/195878436-3c13f8ba-dd70-4e8e-ae0b-70d7097b2f4a.png">

**Reply by Lyraeone** [reply]
I had 3 initially and bought 3 more recently.
These are the versions:

2020-09-23 13:52
2020-09-23 17:45
2020-09-24 13:26
2020-09-24 13:48
2021-12-21 10:25
2021-12-21 10:32

**Reply by sjorge** [reply]
Are you able to run the dev branch? I wrapped the read of the 2 new attributes in a try/catch so if they fail say due to a timeout the entire (re)configure would not fail.

If you are running that version I'd be interest in a debug log of a read attempt of `led_control` for a 2021 and 2020 firmware version.

You should be able to hit the refresh icon in the webui:
<img width="797" alt="image" src="https://user-images.githubusercontent.com/379665/196035124-e8dd37b5-798b-4a32-8bb7-9deaa32765d6.png">

**Reply by Lyraeone** [reply]
I'm not sure on how to run the DEV branch. I did try some reconfigure on both version 2020 and 2021. I get these errors. The Bathroom is the 2021 version:

Error 2022-10-16 15:53:08Failed to configure '0-3-MS-Frontdoor', attempt 4 (Error: Bind 0x0015bc001a01a8b1/35 genPowerCfg from '0x00212effff07515f/1' failed (waiting for response TIMEOUT) at DeconzAdapter.bind (/app/node_modules/zigbee-herdsman/src/adapter/deconz/adapter/deconzAdapter.ts:768:19) at Endpoint.bind (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:567:13) at Object.bind (/app/node_modules/zigbee-herdsman-converters/lib/reporting.js:40:9) at Object.configure (/app/node_modules/zigbee-herdsman-converters/devices/develco.js:520:13) at Configure.configure (/app/lib/extension/configure.ts:115:13) at Configure.onMQTTMessage (/app/lib/extension/configure.ts:55:21))
Error 2022-10-16 15:53:25Failed to configure '1-4-MS-Bathroom', attempt 4 (Error: Bind 0x0015bc001a022a0a/35 genPowerCfg from '0x00212effff07515f/1' failed (waiting for response TIMEOUT) at DeconzAdapter.bind (/app/node_modules/zigbee-herdsman/src/adapter/deconz/adapter/deconzAdapter.ts:768:19) at runMicrotasks (<anonymous>) at runNextTicks (node:internal/process/task_queues:61:5) at processTimers (node:internal/timers:499:9) at Endpoint.bind (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:567:13) at Object.bind (/app/node_modules/zigbee-herdsman-converters/lib/reporting.js:40:9) at Object.configure (/app/node_modules/zigbee-herdsman-converters/devices/develco.js:520:13) at Configure.configure (/app/lib/extension/configure.ts:115:13) at Configure.onMQTTMessage (/app/lib/extension/configure.ts:55:21))

**Reply by Lyraeone** [reply]
When I try to set the led_control from the GUI, it gives these same messages for both:

Error 2022-10-16 15:57:51Publish 'set' 'led_control' to '1-4-MS-Bathroom' failed: 'Error: Write 0x0015bc001a022a0a/35 genBasic({"develcoLedControl":255}, {"sendWhen":"active","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4117,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUP_MANUF_GENERAL_COMMAND')'
Error 2022-10-16 15:58:00Publish 'set' 'led_control' to '0-0-MS-Frontdoor' failed: 'Error: Write 0x0015bc001a01aa16/35 genBasic({"develcoLedControl":2}, {"sendWhen":"active","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4117,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUP_MANUF_GENERAL_COMMAND')'

I have not yet removed and rejoined the device after pulling the latest Zigbee2MQTT version. I will be ordering some more of these, so I could see what happens then.

**Reply by sjorge** [reply]
> I'm not sure on how to run the DEV branch. I did try some reconfigure on both version 2020 and 2021. 
https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html

Was the device awake when you triggered the reconfigure? It's hitting a timeout on configuring the genPowerCfg bindings to setup reporting of the battery voltage.

**Reply by Lyraeone** [reply]
I'm sure it's awake because when I walk by the Occupancy is changing in the Zigbee2MQTT Gui from Clear to Occupied. Also the LED is flashing. This one is tied into a home assistant automation that turns on the HUE lights 2 hours after sunset on motion and that also still works.

I run a docker on my PI so I will have a look at running the DEV branch in the upcoming days. I'm still a beginner when it comes to this :)

**Reply by Lyraeone** [reply]
The version before October did not give me any errors on repairing or reconfiguring by the way, as I just recently set the whole thing up in September due to my SD card failing a year after using it...

**Reply by sjorge** [reply]
Yeah in the october release support for writing to the develco specific occupancy_timeout and led_control got added, as wel as batteryVoltage reporting. This was tested on a MOSZB-140 I ordered to see if I could add support for the missing features mentioned.

I'm really surprise you are hitting timeouts even on batteryVoltage reporting. I wonder if the device only has a brief window after poweron or after a reset where it accepts the reporting configuration.

**Reply by dermarkusauslb** [reply]
Hello together. 

I am following your discussion here as I got myself a MOSZB-140 with firmware 3.6.6 as of 2021-12-21 19:30.

My WebGUI (1.28.4-1) doesn't list the led-switch. After having seen a link on the code I realized this is only display if firmware >=4.x.. is that right ? 

I got a 4.x firmware from frient and they say OTA would be possible. I wonder how I can upgrade the firmware in order to use the led-switch button section in your screenshot above (https://github.com/Koenkk/zigbee2mqtt/discussions/14178#discussioncomment-3889007) 

I want to switch LED off for sure :-)

thanks and regards
Markus

**Reply by sjorge** [reply]
Correct, you need a newer firmware. If you have the firmware file it's pretty easy to include it so z2m can update it.
I recently did this for their airquality sensors.

https://github.com/Koenkk/zigbee-OTA has instructions on how to add firmware images to the repo.
Here is the PR I did for the AQ one so yours should look similar to this one https://github.com/Koenkk/zigbee-OTA/pull/176/files

**Reply by dermarkusauslb** [reply]
hej sjorge,

thank you for your fast and helpful answer.

sorry for asking, but I checked both your tips and I already found this firmware in the repo (index.json, lines 2230++), it is mentioned there already. It is the exact version frient sent me. 

I am quite stuck what to do now - Can you give me another hint ? Sorry, I am new to this and my only aim is to switch the leds off...

Meanwhile I will try to figure out how far I come...

thank you - again -
Markus

**Reply by sjorge** [reply]
I just made a PR for this https://github.com/Koenkk/zigbee-herdsman-converters/pull/5151, once this gets merged, you should switch to the dev branch, or wait for a new z2m release. You should then be able to upgade the firmware.

**Reply by dermarkusauslb** [reply]
Thank you
